docs/enhancements/fep-002-config-hooks.md

```diff
@@ -0,0 +1,204 @@
+that such configuration needs to be updated/revoked/etc.
+
+## Motivation
+
+Integration with external systems when devices may need specific
+per-device tokens, per-device configurations sometimes transient
+is a not-well covered use case in our API. Enabling such thing
+with today's API would not be possible. This is why we propose
+this configuration hooks API.
+
+This could be useful i.e. to integrate with ACM, where a service could
+be running on the ACM cluster, and the fleets configured to have
+the devices onboarded to ACM would point to that service in the fleet
+configuration items to fetch the configuration at specific
+points in time (hook points), anyway this is just an example, the
+mechanism should be generic to enable customers to leverage this.
+
+### Goals
+
+* Define hook points in the device configuration or flightctl
+  to enable the described functionality.
+
+* Allow an external system to be queried to retrieve per-device specific
+  configuration.
+
+* Allow the external system to call-back to flightctl to inform about
+  changes to be made in the previously supplied configuration.
+
+### Non-Goals
+
+* Define a specific product integration, this is a generic mechanism
+  that could be used with any external system.
+
+## Proposal
+
+We define a specific type of configuration that can be used to query an external system.
+
+```yaml
+apiVersion: v1alpha1
+kind: Fleet
+metadata:
+  creationTimestamp: "2024-04-30T14:06:17Z"
+  generation: 1
+  labels: {}
+  name: optical-inspector-production-fleet
+  owner: ResourceSync/basic-nginx-r3sourcesync
+spec:
+  selector:
+    matchLabels:
+      device: optical-inspector
+  template:
+    metadata:
+      generation: 1
+    spec:
+      config:
+...
+...
+      - configType: ExternalConfigProviderSpec
+        externalEndpoint:
+          config-name: my-config
+          url: https://acm-cluster-flightctl-hook.example.com/flightctl-hooks/
+          httpCredentials: my-credentials
+...
+...
+
+      os:
+        image: quay.io/flightctl/flightctl-agent-basic-nginx:latest
+      monitoring:
+        systemd:
+          matchPatterns:
+          - microshift.service
+          - crio.service
+          - flightctl-agent.service
+status:
+  conditions:
+  - lastTransitionTime: "2024-05-09T14:20:21Z"
+    status: "False"
+    type: OverlappingSelectors
+```
+
+### High level format of the hook calls
+#### FlightCTL to external system
+
+##### CALL: POST https://acm-cluster-flightctl-hook.example.com/flightctl-hooks/
+[{"event: "ON_DEVICE_SPEC_UPDATE",
+ "event_uuid": "uuid-for-tracking",
+ "config_-_name": "my-config",
+ "device_metadata": { -json representation of the device metadata on our API, including the owner/fleet -},
+ "callback_endpoint": "https://api.flightctl.example.com/external-provider-callback/",
+ "callback_token": "my-token"}
+ "last_config_hash": "hash-of-the-last-applied-config"
+]
+
+###### RESPONSE: 200 OK
+ [{"event": "ON_DEVICE_SPEC_UPDATE",
```

Comment:
This shows it's a query (resquest/response), not an event notification. So then we can have this handled by the Repository task.